Sources/Service/Environment/Environment+DotEnv.swift

```diff
@@ -0,0 +1,76 @@
+
+import Core
+#if os(Linux)
+    import Glibc
+#else
+    import Darwin
+#endif
+
+extension Environment {
+    /// Loads the ".env" and the ".env.environment" files if they exist
+    /// Both a generic .env file and an environment specific file are loaded.
+    /// This allows you to set custom credentials to specific environments such as
+    /// production, testing, and development.
+    /// If you use custom environments, those will also be loaded.
+    /// Note: The .env files must be located in the root directory.
+    public func load() {
+        // Load the default one
+        self.load(environment: ".env")
+        
+        // Load any environment specific ones next
+        self.load(environment: ".env.\(self.name)")
+    }
+    
+    /// Loads the specific env file and sets the appropiate environment variables
+    /// Parse the contents of the .env file into a readable key/value
+    /// - parameter environment: The filename of the .env file to load. i.e. `.env`.
+    /// Note: The .env files must be located in the root directory.
+    public func load(environment fileName: String) {
+        // Create the file path
+        let url = URL(fileURLWithPath: DirectoryConfig.detect().workDir).appendingPathComponent(fileName)
```

Comment:
The `DirectoryConfig` should be supplied here, or even better, created from services. That way the user can configure a custom working directory.